test/specs/common/app.spec.js

```diff
@@ -100,6 +101,11 @@ describe("App", () => {
     });
   });
 
+  it("should allow all possible valid characters for NPM scope and NPM package name", () => {
+    let npmScopeAndPackageName = "@company-name~test_se.connect/app-name_connect~carrier.app";
+    expect(npmScopeAndPackageName).to.match(regex.appName);
```

Comment:
I like the creative use of match here!

Reply:
Yeah it's super handy, I considered using `to.deep.equal(RegExp, "should match")` and then I remembered mocha provides `.match(RegExp)`.